### PR TITLE
PN-3013: Add looping

### PR DESCRIPTION
**Pull request by timidri**
This PR adds looping to the workflow-run step.
The step implementation has also been converted to Python. 
This PR is blocked by [this PR](https://github.com/puppetlabs/relay-sdk-python/pull/6) which adds the workflow run method to `relay-sdk-python`.

### CODE REVIEW

File: steps/run/step.py
Comment (kyleterry) [inline steps/run/step.py]: You can probably remove this comment. Limits and throttling are always implemented on the API side since adhoc and 3rd party clients can burst requests anyway.

File: steps/run/step.py
Comment (kyleterry) [inline steps/run/step.py]: I don't see the decorators for the workflow run link being set like they are in the shell script. Are these being removed?

File: steps/run/test-metadata.yaml
Comment (impl) [inline steps/run/test-metadata.yaml]: I don't think we necessarily want to include these files here, at least until we figure out what a testing framework might look like.

File: steps/run/step.yaml
Comment (impl) [inline steps/run/step.yaml]: Using long unquoted strings in YAML gets really weird (like you can't use ":" characters), so I'd suggest making this a multi-line form using `>` instead.

File: steps/run/step.py
Comment (impl) [reply steps/run/step.py]: I don't think it was intentional to remove the decorators, so we should figure out how to get them back.

File: steps/run/spec.schema.json
Comment (impl) [inline steps/run/spec.schema.json]: I spent a bit of time thinking about this name after our discussion last week -- might I suggest calling this `parameterSets` instead (note most steps use camelCase)? I'm afraid it won't be clear here that each item of the parameters array will be a separate invocation, instead of just passing all the parameters to one run (even though that isn't technically possible).

File: steps/run/Dockerfile
Comment (impl) [inline steps/run/Dockerfile]: Tiny nit, but I don't think you need to call out the implementation of the step really. We don't currently use this label anywhere but I think it might be confusing to see when it doesn't change the functionality.

**Review by impl — COMMENTED** [review]
A few points of hopefully small feedback for you - this will be a game changer once it lands!

File: steps/run/step.py
Comment (kyleterry) [reply steps/run/step.py]: Agreed! I don't want to see that feature get forgotten about. I'm still waiting for the frontend side of it :D.

File: steps/run/spec.schema.json
Comment (timidri) [reply steps/run/spec.schema.json]: Done!

File: steps/run/step.py
Comment (timidri) [reply steps/run/step.py]: @kyleterry @impl I totally forgot about decorators since I wasn't able to find docs on what they were. Are there any? 
I can't re-add decorators right now, but I can do that if I know what they do and I also would need to change the relay-python-sdk to return the response from the API. I suggest we raise a ticket for this which I can pick up after finalizing this PR.

File: steps/run/step.py
Comment (impl) [inline steps/run/step.py:10]: You could probably reduce the duplication of the logic below by just making the one-run case a special case of `param_sets`, i.e., instead of `params = relay.get(D.parameters)`, say `param_sets = [relay.get(D.parameters)]`. (Then change the log message based on `len(param_sets)` or whatever is convenient, as I think it's the only thing that's actually different between those code paths.)

File: steps/run/step.py
Comment (impl) [inline steps/run/step.py]: Same here regarding E501 as the other PR - just move this comment to the line above.

**Review by impl — APPROVED** [review]
LGTM, pending merge of corresponding SDK PR.

**Pull request merged**
